app/ui-message/client/message.js

```diff
@@ -177,6 +177,14 @@ Template.message.helpers({
 			return msg.avatar.replace(/^@/, '');
 		}
 	},
+	customAvatarUsername() {
```

Comment:
The fact is you just need to apply the new validation on Livechat rooms, otherwise, you need to keed the current logic.
Due to that, I suggested implementing the new validation based on `RoomTypes` instead of a hardcoded solution.

Reply:
This was my current aproach:-

getUserAvatar() {
		const { room } = this;

		return roomTypes.roomTypes[room.t].getAvatarPath(room);
	},

Reply:
If we make the custom function for livechat rooms only then we have to rely on the `username=msg.u.username` for non livechat rooms

Reply:
And other approach can be to make this custom function for all the roomtypes

Reply:
@renatobecker Which should be better approach? Please suggest.

Reply:
In this scope `this` is the message not the `room`.

Reply:
`this` contains message, settings, room in this scope

Reply:
In message.js its used like :- 
const { msg, room = {}, u } = this;

Reply:
Yeah, but you need to focus on the `message` payload, is from there you will extract the information you need to deal/return the username/url.

Reply:
Right @renatobecker :+1:. So what should be our approach now

Reply:
Yes, this is my suggestion, create a custom function to return the correct avatar URL.